starknet_py/net/full_node_client.py

```diff
@@ -422,7 +425,7 @@ def _create_broadcasted_txn(transaction: Union[Invoke, Declare, DeployAccount])
 
 
 def _create_broadcasted_declare_properties(transaction: Declare) -> dict:
-    contract_class = transaction.dump()["contract_class"]
+    contract_class = cast(Dict, DeclareSchema().dump(obj=transaction))["contract_class"]
```

Comment:
Why do we cast here?

Reply:
Because pyright complains :/ the returned value is a dict, but it sees a list. I can add `ignore`

Reply:
I prefer to cast than to ignore in this case.